src/TraceEvent/TraceEventSession.cs

```diff
@@ -1368,6 +1368,35 @@ public static unsafe List<string> GetActiveSessionNames()
             return activeTraceNames;
         }
 
+        /// <summary>
+        /// Get the maximum number of ETW loggers supported by the current machine
+        /// </summary>
+        /// <returns>The maximum number of supported ETW loggers</returns>
+        private static int GetETWMaxLoggers()
+        {
+            const string MaxEtwRegistryKey = "SYSTEM\\CurrentControlSet\\Control\\WMI";
+            const string MaxEtwPropertyName = "EtwMaxLoggers";
+            const int DefaultMaxETWLoggers = 64;
+
+            try
+            {
+                using (RegistryKey key = Registry.LocalMachine.OpenSubKey(MaxEtwRegistryKey))
+                {
+                    if (key != null)
+                    {
+                        if (int.TryParse(key.GetValue(MaxEtwPropertyName).ToString(), out int maxEtwLoggers))
```

Comment:
The registry key is not present by default.  Can you check to see if the key and value are present first before trying to use them?  As is, I would expect this to throw in the common case when calling `GetValue().ToString()`, since the result will likely be `null`.

Reply:
I have added a check to see if the value is present alongside the key before using.